NullPointerException while getting the SSLContext

Hi,

Launching a _parallel_ query to multiple WSClients ends up with an NPE:
`Caused by: java.lang.NullPointerException
    at com.vmware.vim25.ws.WSClient.getTrustAllSocketFactory(WSClient.java:225)
    at com.vmware.vim25.ws.WSClient.<init>(WSClient.java:85)
    at com.vmware.vim25.ws.WSClient.<init>(WSClient.java:69)
    ... 23 more`
`2016-06-20 11:10:44,469 ERROR query                    : Task failed in daemon "query"
java.lang.NullPointerException
    at com.vmware.vim25.mo.ServiceInstance.constructServiceInstance(ServiceInstance.java:121)
    at com.vmware.vim25.mo.ServiceInstance.<init>(ServiceInstance.java:104)`

By looking at _TrustAllSSL.getTrustContext()_ method I guess that the 'problem' comes from the fact that the _setAlreadyCreated()_ call is done too early; Making the other threads 'think' that the context was already created, which is not the case.

Making the getTrustContext method synchronized could do the job, or maybe just delaying the   setAlreadyCreated() call. Any help would be appreciated! 

Here's my environment:
Java: 1.8.0_66-b17
yavijava: 6.0.04
vSphere: 6.0.0

Thanks!
Alex

--- a/src/main/java/com/vmware/vim25/ws/TrustAllSSL.java
+++ b/src/main/java/com/vmware/vim25/ws/TrustAllSSL.java
@@ -27,7 +27,7 @@ public class TrustAllSSL {
     private static boolean alreadyCreated = false;
     private static SSLContext sslContext;
 
-    public static SSLContext getTrustContext() throws RemoteException {
+    public static synchronized SSLContext getTrustContext() throws RemoteException {
         try {
             if (getAlreadyCreated()) {
                 return sslContext;

